### PR TITLE
py sol for swap

### PR DESCRIPTION
**Pull request by freakin23**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [ ] I have linked this PR to any issues that it closes.

### CODE REVIEW

File: solutions/gold/ac-swap.mdx [hunk 1]
```diff
@@ -169,6 +169,90 @@ int main() {
...
+
+	if s[i] == "K":
+		pos[0].append(i)
+	elif s[i] == "E":
+		pos[1].append(i)
+	else:
+		pos[2].append(i)
+
+dp = [
+	[[[0] * (count[2] + 1) for _ in range(count[1] + 1)] for _ in range(count[0] + 1)]
```
Comment (TheGamingMousse) [inline solutions/gold/ac-swap.mdx:210]: this is a little hard to read. could you replace it with something a bit more intuitive?

**Review by TheGamingMousse — COMMENTED** [review]
at first glance it looks pretty good to me

File: solutions/gold/ac-swap.mdx (hunk 1 above)
Comment (TheGamingMousse) [reply solutions/gold/ac-swap.mdx:210]: ok its not that bad but at least to me it does look slightly confusing

Comment (SansPapyrus683) [reply solutions/gold/ac-swap.mdx:210]: ^

Comment (freakin23) [reply solutions/gold/ac-swap.mdx:210]: it's 4D, I don't know any other way to initialize it other than list comprehension. I can add a comment to it like:
# dp[cur][ks][es][ys] = number of ways to achieve the state with ks 'K's, es 'E's, ys 'Y's used, and cur swaps made

**Review by ryanchou-dev — APPROVED** [review]
woah it's been so long since i wrote this sol

lgtm 👍

**Review by SansPapyrus683 — APPROVED** [review]

**Pull request merged**